opm/simulators/linalg/BlackoilAmg.hpp

```diff
@@ -553,6 +554,13 @@ class OneStepAMGCoarseSolverPolicy
     CoarseLevelSolver* createCoarseLevelSolver(LTP& transferPolicy)
     {
         coarseOperator_=transferPolicy.getCoarseLevelOperator();
+	if( param_->cpr_solver_verbose_ >10 ){
+	    std::string filename = "course_matrix_istl_blackoil.txt";
+	    std::ofstream filem(filename);
+	    //const auto& matrix = *transferPolicy.getCoarseMatrix();
+	    const auto& matrix = coarseOperator_->getmat(); 
+	    Dune::writeMatrixMarket(matrix, filem);
```

Comment:
Like proposed in another comment we could do a reinterpret_cast to `BCRSMatrix<FieldMatrix>` here.

Reply:
I think if this workes it wold be nice. It is just messy and a copy of the code from matrixmarket.hh. But it is very convenient to be able to write out the matrices from OPM.